OpenDreamRuntime/Objects/MetaObjects/DreamMetaObjectClient.cs

```diff
@@ -34,22 +33,22 @@ sealed class DreamMetaObjectClient : IDreamMetaObject {
 
             switch (varName) {
                 case "eye": {
-                    string ckey = dreamObject.GetVariable("ckey").GetValueAsString();
-                    DreamObject eye = value.GetValueAsDreamObject();
-
-                    //Runtime.StateManager.AddClientEyeIDDelta(ckey, eyeID);
+                    /*if(dreamObject.GetVariable("ckey").TryGetValueAsString(out var ckey) && variableValue.TryGetValueAsDreamObject(out var eye)) {
+                        Runtime.StateManager.AddClientEyeIDDelta(ckey, eyeID);
+                    }*/
                     break;
                 }
                 case "perspective": {
-                    string ckey = dreamObject.GetVariable("ckey").GetValueAsString();
-
-                    //Runtime.StateManager.AddClientPerspectiveDelta(ckey, (ClientPerspective)variableValue.GetValueAsInteger());
+                    /*if(dreamObject.GetVariable("ckey").TryGetValueAsString(out var ckey) && variableValue.TryGetValueAsInteger(out var perspective)) {
+                        Runtime.StateManager.AddClientPerspectiveDelta(ckey, (ClientPerspective) perspective);
+                    }*/
                     break;
                 }
                 case "mob": {
-                    DreamConnection connection = _dreamManager.GetConnectionFromClient(dreamObject);
-
-                    connection.MobDreamObject = value.GetValueAsDreamObject();
+                    DreamConnection? connection = _dreamManager.GetConnectionFromClient(dreamObject);
+                    if (connection != null && value.TryGetValueAsDreamObject(out var valueDreamObject) && valueDreamObject != null) {
```

Comment:
`client.mob = null` is a valid operation.
```suggestion
                    if (connection != null && value.TryGetValueAsDreamObject(out var valueDreamObject)) {
```